specification/app/Microsoft.App.DynamicSessions/routes.tsp

```diff
@@ -0,0 +1,98 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "./main.tsp";
+import "./models/code-execution.tsp";
+import "./models/code-execution-file.tsp";
+import "./models/common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+namespace Microsoft.App.DynamicSessions;
+
+interface CodeExecution {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" ""
+  @doc("Execute code in a session.")
+  @route("/executions")
+  @pollingOperation(CodeExecution.getResult)
+  executeCode is Foundations.Operation<
+    {
+      ...SessionIdentifier;
+      ...ExecutionOperationIdHeader;
+
+      @doc("The request to execute code.")
+      @body
+      codeExecutionRequest: SessionCodeExecutionRequest;
+    },
+    AcceptedResponse &
+      SessionCodeExecutionResponse &
+      ExecutionOperationIdHeader &
+      Foundations.LongRunningStatusLocation
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" ""
+  @doc("Get the code execution result.")
+  @route("/executions/{executionId}")
+  getResult is Foundations.Operation<
+    {
+      ...SessionIdentifier;
+      ...CodeExecutionIdentifier;
+    },
+    {
+      ...SessionCodeExecutionResponse;
+    }
+  >;
+}
+
+alias SessionResourceFilesTraits = NoRepeatableRequests &
+  NoConditionalRequests &
+  NoClientRequestId &
+  QueryParametersTrait<{
+    ...SessionIdentifier;
+    ...SessionResourceFilePathQueryParameter;
+  }>;
+
+alias SessionResourceFilesOperations = Azure.Core.ResourceOperations<SessionResourceFilesTraits>;
+
+interface SessionResourceFiles {
+  @doc("List the file resources.")
+  listFilesMetadata is SessionResourceFilesOperations.ResourceList<SessionResourceFile>;
+
+  @doc("Get the file resource.")
+  getFileMetadata is SessionResourceFilesOperations.ResourceRead<SessionResourceFile>;
+
+  @doc("Delete the file.")
+  deleteFile is SessionResourceFilesOperations.ResourceDelete<SessionResourceFile>;
+
+  #suppress "@azure-tools/typespec-azure-core/byos" ""
+  @doc("Upload a file to a session.")
+  @route("/files")
+  uploadFile is RpcOperation<
+    {
+      @doc("The content type for the operation. Always multipart/form-data for this operation.")
+      @header("content-type")
+      contentType: "multipart/form-data";
+
+      @doc("The file to upload.")
+      file: bytes;
+    },
+    SessionResourceFile,
+    SessionResourceFilesTraits
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" ""
+  @doc("Get the content of the file.")
+  @route("/files/{name}/content")
+  getFileContent is Foundations.Operation<
```

Comment:
This fits the ResourceAction perfectly. See [playground](https://azure.github.io/typespec-azure/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3JlIjsKCnVzaW5nIFR5cGVTcGVjLkh0dHA70BVSZXN00RVWyXLIG0HEWS5Db3Jl0hIuVHJhaXRzOwoKQHVzZUF1dGgoCiAgQXBpS2V5xA48xgtMb2NhdGlvbi5oZWFkZXIsICJhcGkta2V5Ij4gfCBPxCoyxS9bCiAgICB7xQYgIOQA3DrHH0Zsb3fkAKUuaW1wbGljaXQsxyVhdXRob3JpesVhVXJsOiAi5AFoczovL2xvZ2luLmNvbnRvc28uY29tL2NvbW1vbi9vxDUyL3YyLjAvyEBlIshSc2NvcGVzOiBbyUl3aWRnZXTNSi5kZWZhdWx0Il3GN30KICBdPgopCkBzZXJ2aWNlKOQAxnRpdGxlOiAiQ8Y6IFfFSSBNYW5hZ2VyIiwKfccxZXLkATkie2VuZHBvaW50fcd05QCV0D9BUElzxRnEYyAgLyoqIApTdXDkAh5lZMglU%2BYAh3MgyFNzIChwcm90b2NvbCBhbmQgaG9zdG5hbWUsIGZvciBleGFtcGxlOgrpAPRlc3R1cy5hcGku8gD%2FKS4KICovxX7IXzogc3RyaW5n5ACYfeQBCOcCvGVkKOcAvy7mAJrnAP3oAmpzKQrkAIZzcGFjZSDVKjsKCuQA41RoySDvAUgg5wFzIMdzLuQAnGVudW0gyGTlASrERccRIDIwMjItMDgtMzHELCAg5AK%2FRGVwZW5kZW5jeSjrAuLIQy52MV8wX1ByZXZpZXdfMikKICBgyUYwYCwKfQoKLy8gTW9kZWxzIC%2FTAQpAcmVzb3VyY2UoImV4ZWN15AKycyIpCm3ENCBTZXPEaUNvZGVFyB7lALhAdmlzaWJpbGl0eSgicmVhZCLEfUBkb2MoIsc1IGNvZGUgyVUgaWQuxiVrZXnLbUnGO2lk6AGtOwrJSuQBX2lkZW50aWZpZXIgb2YgdGhlIHPGYMZOyh%2FXQMlp5QO7yETvAJxyZXF1ZXN0xlPJF%2BQD3NdWc3RhdHVz105vcGVy5QP7LCBpbmRpY2F0ZXMgd2hldGjkAhh1Y2NlZWRlZCBvciBub8d0xlA6IEZvdW5kxTxzLk%2FIR1N0YXRl8ADXcnJv5wESaXPxAWtmIGZhaWxl5wFyxSw%2Fzl1FxBRSZXNwb25z0FxyZXN1bHTYXekA3sZdxi0%2FOvUCLFLFHTvkAojWX%2FABPMRU%2BgJ7xk%2FmAoHJT3N0YW5kYXJkIG91dHB121jkAWZkb3XkAKz4AdTGSesBXNlIZXLkAVLLSCNzdXBwcmVzc%2F8HDW9yZS9uby11bmtub3duIiAiIvkBfNF95QRw6gLHaXMgZmllbGQgaXMgc2FtZSBhc8U2yCJhY3R1YWzdUyBhZnRlciBiZeQHREpzb27kBLhpYWxpeukCPe4BfT86IOcAu%2FoDaGlt7gNoaW4gbWlsbGlzZWNvbmRzz1hUaW1lSW5NyyE%2FOiBpbnQ2NO8COG9wxC9hbMtraWQg5ge%2F6gI46QIt6QM%2FSWRIxSTmAjjGMO8EY%2BkBUPkC58kPSWTrAfjpAKDoBPTrBK5xdWVyeSBwYXJhbWV07ACixyxJyivuAtF1c2VyLWFzc2lnbmVk%2FwUAIEDFZ8QJbWluTGVuZ3RoKDHFGW1heMcQMjU2xRJwYXR0ZXJuKCJeW0EtWmEtejAtOS1fXXsxLDI1Nn0k%2BQVQfQoKYWxpYXPqAVzmCWMgPSBRxH5Q6ADkc8UYPPEA4T47x0HpAZTEPOsG0VLnBoXKIDzPbD47CgppbnRlcmbkB5TJHeYHRmNyZWF0ZeQDB8pBLkxvbmdSdW7kCjfIX0PFKFdpdGjnCIpQcm92aWRlZE5hbWU85Qg79AR35glO6QD0UuYGLeYCVecA8%2FoCdj4gJsU%2FICDoBW7fOck5CiAg5AD2ICDkCFfuANzoANFSZWFk6AFv7QCzxjkvLyBZb3UgY2FuIOYCey9maWxlL3vkCN19L%2BQJPGVudOQCGmFjxDUoIscT5gJaxhVTZeQCw3RvcigiL8YYZ2XENGdldOQI%2BmVu%2BACcQcU%2F%2FwFWIHt9xwjFBiAg5wCCOiBieXRlczvpCsg%2B5AKH&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fdata-plane%22%5D%7D%7D)

Reply:
Thank for the comments and sample code in playground, I am able to update the original GetExecutionResult to use standard operation.

But for execute code, the request body is model `SessionCodeExecutionRequest` and is different from the response `SessionCodeExecutionResource`, so `LongRunningResourceCreateWithServiceProvidedName` might not be a perfect fit in this case.

And for GetFileContent, using ResourceAction would let this Api become a POST operation rather than a GET, so I was't able to update to use ResourceAction.

Are there any suggestions for this scenario? Thanks!

Reply:
Actually the playground countains another example for the `getContent` using `ResourceAction`. 

I pulled your PR and used following code and it produced same swagger
```
  #suppress "@azure-tools/typespec-azure-core/verb-conflict" "GET allowed on Action"
  @action("content")
  @actionSeparator("/")
  @doc("Get the content of the file.")
  @get
  getContent is SessionResourceFilesOperations.ResourceAction<
    SessionResourceFile,
    {},
    {
      @body _: bytes;
    }
  >;
```
Please also remove now unused `model SessionResourceFileNamePathParameter` from `models/common.tsp`.

Reply:
I will remove -1 on the review as I will be OOF tomorrow. But please do fix this.

Reply:
Thank you, this is fixed now